### PR TITLE
Add CUDA EP optimization option to Stable Diffusion examples

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
This PR adds `--provider` option to the stable diffusion example so that the user can optimize the model for cuda ep instead if desired. 
The directml ep and settings are still the default so the interface and outputs does not change at all for it. 

In a follow-up PR, I will add an example test for the cuda ep if the memory and time requirements for the stable diffusion (not xl) example fits on our gpu pool. 

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
Stable Diffusion and Stable Diffusion XL models can now be optimized for CUDA EP using `--provider cuda` option.

### CODE REVIEW

File: examples/directml/stable_diffusion/stable_diffusion.py [hunk 1]
```diff
@@ -334,24 +367,34 @@ def optimize(
             "as expected."
         )
 
-    if version.parse(ort.__version__) < version.parse("1.15.0"):
-        print("This script requires onnxruntime-directml 1.15.0 or newer")
+    if args.provider == "dml" and version.parse(ort.__version__) < version.parse("1.16.0"):
+        print("This script requires onnxruntime-directml 1.16.2 or newer")
```
Comment (PatriceVignola) [inline examples/directml/stable_diffusion/stable_diffusion.py:371]: `1.16.0` or newer

File: examples/directml/stable_diffusion_xl/README.md [hunk 2]
```diff
@@ -74,15 +75,34 @@ If you omit `--interactive`, the script will generate the requested number of im
 
 The minimum number of inferences will be `ceil(num_images / batch_size)`; additional inferences may be required of some outputs are flagged by the safety checker to ensure the desired number of outputs are produced.
 
-# Issues
+## Stable Diffusion Optimization with CUDA
+This example can also be used to optimize Stable Diffusion models for CUDA. You can use the same `stable_diffusion.py` script with `--provider cuda` option to optimize the models for CUDA and test inference. The optimized models will be stored under `models/optimized-cuda/[model_id]` (for example `models/optimized-cuda/runwayml/stable-diffusion-v1-5`).
```
Comment (PatriceVignola) [inline examples/directml/stable_diffusion_xl/README.md:79]: `models/optimized-cuda/runwayml/stable-diffusion-v1-5` is the default path for SD 1.15. We should change the example here for SDXL

File: examples/directml/stable_diffusion/stable_diffusion.py (hunk 1 above)
Comment (jambayk) [reply examples/directml/stable_diffusion/stable_diffusion.py]: Thanks! updated

File: examples/directml/stable_diffusion_xl/README.md (hunk 2 above)
Comment (jambayk) [reply examples/directml/stable_diffusion_xl/README.md]: Thanks! updated

**Review by devang-ml — COMMENTED** [review]
Looks good to me.

**Review by PatriceVignola — APPROVED** [review]

**Comment by PatriceVignola** [comment]
:shipit:

**Pull request merged**